pulsar-broker/src/main/java/com/yahoo/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -573,31 +585,30 @@ private void updateNamespaceBundles(NamespaceName nsname, NamespaceBundles nsBun
         checkNotNull(nsname);
         checkNotNull(nsBundles);
         String path = joinPath(LOCAL_POLICIES_ROOT, nsname.toString());
-        LocalPolicies policies = null;
-        try {
-            policies = this.pulsar.getLocalZkCacheService().policiesCache().get(path);
-        } catch (KeeperException.NoNodeException ne) {
+        Optional<LocalPolicies> policies = pulsar.getLocalZkCacheService().policiesCache().get(path);
+
+        if (!policies.isPresent()) {
             // if policies is not present into localZk then create new policies
-            this.pulsar.getLocalZkCacheService().createPolicies(path, false);
+            this.pulsar.getLocalZkCacheService().createPolicies(path, false).get();
             policies = this.pulsar.getLocalZkCacheService().policiesCache().get(path);
         }
-        policies.bundles = getBundlesData(nsBundles);
+
+        policies.get().bundles = getBundlesData(nsBundles);
```

Comment:
I'm not 100% sure but `policies.get()` can be `null` here if creating new policies failed.